### ISSUE TITLE
Beautify functionality messes up with empty lines

### ISSUE THREAD

**Issue by juh9870** [ayecue/miniscript-vs#4]
The Beautify function inserts empty lines in a middle of long comments, and removes empty lines in between statements

Before Beautify:
```
// Plots a function on a graph
// @param {(number) => number} f - function to plot
// @param {Range} [domain] - domain of x values to plot. Defaults to unlimited
// @param {Color} [plotColor=color.red] - Graph line color. Defaults to color.red
// @param {number} [thickness=2] - Graph line thickness
Graph.plot = function(f, domain = null, plotColor = null, thickness = 2)
```
After:
```
// Plots a function on a graph

// @param {(number) => number} f - function to plot

// @param {Range} [domain] - domain of x values to plot. Defaults to unlimited
// @param {Color} [plotColor=color.red] - Graph line color. Defaults to color.red
// @param {number} [thickness=2] - Graph line thickness

Graph.plot = function(f, domain = null, plotColor = null, thickness = 2)
```

Before:
```


    rangeX = self.rangeX
    rangeY = self.rangeY

    scaleX = self.scaleX
    scaleY = self.scaleY

    bottom = self.bottom
    top = self.top
    left = self.left
    right = self.right

    display = self.display
```
After:
```
	rangeX = self.rangeX
	rangeY = self.rangeY
	scaleX = self.scaleX
	scaleY = self.scaleY
	bottom = self.bottom
	top = self.top
	left = self.left
	right = self.right
	display = self.display
```

It also seems to insert empty lines around all control structures.

I would like to use Beautify as my default formatter, but inserting or removing empty lines seems excessive

EDIT: File I use for testing: https://github.com/juh9870/mini-graph/blob/main/Graph.ms

**Comment by juh9870** [comment]
After this is resolved, would be great to also have the extension provide a code formatter functionality, using Beautify

**Comment by ayecue** [comment]
Yes indeed the beautifier certainly needs some attention. I'll take a look on how I can improve it.

**Comment by juh9870** [comment]
It might be useful to have some sort of tests for Beautifier too, to ensure that it formats everything as intended.

My Miniscript formatter has a bunch of tests, that should be pretty easy to hook up to your Beautifier as a starting point, after all it's a string->string operation

https://github.com/juh9870/miniscript-syntax/blob/master/src/test/suite/extension.test.ts

**Comment by ayecue** [comment]
Well there are tests in it's respective package https://github.com/ayecue/greybel-transpiler/blob/master/tests/transpiler.test.js. Therefore it's not really necessary to add tests within the vscode extension. Even though it would be probably good to add some visual tests but I am lacking time to implement those.

Anyway I didn't really take a lot of time adjusting the beautifier as it didn't seem to have many users anyway since it mainly served the purpose of deobfuscating code. But it's good to see somebody is interested to see the beautifier extended so it's more useful.

**Comment by juh9870** [comment]
I mostly gleaned those from MiniScript default library formatting, wiki pages and asking Joe questions about "preferred" formatting

**Comment by juh9870** [comment]
Oh, my tests are not VSCode-related BTW. They just test formatting in a vacuum, no part of VSCode API is used in there. just fastcheck library for generating random code to test on, and function that takes unformatted code string and spews out formatted code string

**Comment by ayecue** [comment]
@juh9870 I finally had some time to do some updates on the beautifier. Unfortunately I couldn't really use your test cases too much since some of them contained invalid syntax. Anyway I did a lot of changes to the beautify formatting and also removed the behaviour of adding or removing empty lines. I feel like it is in a state where it actually provides more readable code now.

Additionally I've also integrated the beautify with the vscode formatter API.

The changes are not yet merged but you can already check them out in the 0.5.3 branch https://github.com/ayecue/miniscript-vs/pull/23.

**Comment by ayecue** [comment]
The vscode formatter is now available with 0.5.3 merged. Let me know if there are any tweaks that should be done.

**Comment by ayecue** [comment]
Due to inactivity I am going to close this issue for now. Feel free to reopen.
